src/main/scala/mesosphere/marathon/state/RootGroup.scala

```diff
@@ -492,21 +492,30 @@ object RootGroup {
     }
 
   trait NewGroupStrategy {
+    def enforceRoleValue(id: AbsolutePathId): Option[Boolean]
+
     def newGroup(id: AbsolutePathId): Group
   }
 
   object NewGroupStrategy {
 
     object Fail extends NewGroupStrategy {
+      override def enforceRoleValue(id: AbsolutePathId): Option[Boolean] = None
+
       override def newGroup(id: AbsolutePathId): Group = throw new RuntimeException(s"No such group: ${id}")
     }
 
-    def fromConfig(newGroupEnforceRoleBehavior: NewGroupEnforceRoleBehavior): NewGroupStrategy =
-      new NewGroupStrategy {
-        override def newGroup(id: AbsolutePathId): Group = {
-          Group.empty(id, enforceRole = if (id.isTopLevel) newGroupEnforceRoleBehavior.option else None)
+    case class UsingConfig(newGroupEnforceRoleBehavior: NewGroupEnforceRoleBehavior) extends NewGroupStrategy {
+      override def enforceRoleValue(id: AbsolutePathId): Option[Boolean] = {
+        if (id.isTopLevel) {
+          newGroupEnforceRoleBehavior.option
+        } else {
+          None
         }
       }
-  }
 
+      override def newGroup(id: AbsolutePathId): Group =
+        Group.empty(id, enforceRole = enforceRoleValue(id))
+    }
+  }
```

Comment:
could we put `if (id.isTopLevel) newGroupEnforceRoleBehavior.option else None` right here and remove `enforceRoleValue` from the traits and objects above?